.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,75 @@
+name: Docker Build
+
+on:
+  schedule:
+    # Schedule runs run off HEAD commit on master.
+    # Run once a day to keep yunojuno/heroku up to
+    # date with Heroku's base image.
+    - cron:  '* 23 * * *'
+
+  push:
+    branches:
+      - master
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  docker-build-push:
+    name: Build Docker image & push (if master)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+        # Docker Meta handles manipulating various GitHub
+        # Action contexts into sane Docker image tags &
+        # labels. See more below.
+      - name: Instantiate Docker Meta
+        id: docker_meta
+        # Hash is pinned to v1.11.0 for security.
+        uses: crazy-max/ghaction-docker-meta@9ae6899cfa6d691ad29d6fc0a228f276a16f5d44
```

Comment:
If we're pinning - does it make sense to fork and pin to our own version so we know we are in control?

Reply:
This is [GitHub's policy on that](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions); basically an attacker would have to generate a SHA-1 collision with their payload for this to be insecure. GitHub have deprecated the short-hash version (`9ae68`) for this reason.

I have only pinned this third-party action because it's from a sole user's account rather than a more trustable organisation (such as GitHub themselves, or Docker).

It's up to us basically - not using the tagged versions of actions across the board creates a significant amount of maintenance overhead.